### DISCUSSION TITLE
Is there a way to limit panning?

### DISCUSSION THREAD

**Discussion by arzatskis** [xyflow/xyflow#2596] [Q&A]
Hi, I'm new to ReactFlow, checked issues / discussions and was unable to find answers.

Is there to control how much you can pan? What I'm trying to achieve is limit panning so nodes always stay visible and can't be dragged outside the view.

Maybe I've missed some setting, or I should dynamically set `translateExtent` prop etc.? But then not sure how to get bounds or currently drawn shapes. 

Any pointers where to look / how to achieve that would be great

**Comment by sagunji** [comment]
@arzatskis did you get any solutions? I am looking to implement something similar.

**Reply by JanPfenning** [reply]
i was trying the same thing. i want the panning to only work on top, right and bottom. Similarly to the [Drag and Drop ](https://reactflow.dev/examples/interaction/drag-and-drop)creation of nodes i wanted to Drag and Drop remove nodes by dragging them off the canvas to the sidebar (left in my case).

**Comment by Fa1z-khan** [comment]
Im looking for the answer to the same, i only want to scroll it vertically and the panning is infinite is there a way to limit the viewport height
